app/src/main/java/com/github/epfl/meili/photo/CameraActivity.kt

```diff
@@ -0,0 +1,283 @@
+                    true
+                }
+                else -> false
+            }
+        }
+    }
+
+    /** Sets up camera */
+    private fun startCamera() {
+        val cameraProviderFuture = ProcessCameraProvider.getInstance(this)
+
+        cameraProviderFuture.addListener({
+            cameraProvider = cameraProviderFuture.get() // Guaranteed to exist
+
+            // Select lensFacing depending on the available cameras
+            lensFacing = when {
+                hasBackCamera() -> CameraSelector.LENS_FACING_BACK
+                //hasFrontCamera() -> CameraSelector.LENS_FACING_FRONT // not working yet
+                else -> throw IllegalStateException("Back and front camera are unavailable")
+            }
+
+            // Setup Preview use case --> to display the preview to the screen
+            preview = Preview.Builder().build()
+
+            // Setup Image Capture use case --> allows user to take photos
+            imageCapture = ImageCapture.Builder()
+                .setCaptureMode(ImageCapture.CAPTURE_MODE_MINIMIZE_LATENCY)
+                .build()
+
+            // Setup which camera to select (default is back)
+            cameraSelector = CameraSelector.Builder()
+                .requireLensFacing(lensFacing)
+                .build()
+
+            // Display preview in activity
+            val previewView = findViewById<PreviewView>(R.id.camera_preview)
+            preview.setSurfaceProvider(previewView.surfaceProvider)
+
+            // Attach the lifecycle of the camera to the activity's lifecycle
+            camera = cameraProvider.bindToLifecycle(
+                this as LifecycleOwner, cameraSelector, preview, imageCapture
+            )
+
+        }, ContextCompat.getMainExecutor(this))
+    }
+
+    /** Takes a picture */
+    private fun takePhoto() {
+        if (imageCapture == null) {
+            return
+        }
+
+        // Create time-stamped output file to hold the image
+        val photoFile = File(
+            outputDirectory,
+            SimpleDateFormat(
+                FILENAME_FORMAT, Locale.US
+            ).format(System.currentTimeMillis()) + ".jpg"
+        )
+
+        // Set up behaviour for when a photo is taken
+        imageCapture!!.takePicture(
+            ImageCapture.OutputFileOptions.Builder(photoFile).build(),
+            ContextCompat.getMainExecutor(this),
+            object :
+                ImageCapture.OnImageSavedCallback { // object with callback functions for pictures
+                override fun onError(exc: ImageCaptureException) {
+                    Log.e(TAG, "Photo capture failed: ${exc.message}", exc)
+                }
+                override fun onImageSaved(output: ImageCapture.OutputFileResults) {
+                    val intent = Intent(applicationContext, PhotoDisplayActivity::class.java)
+                    intent.putExtra(URI_KEY, Uri.fromFile(photoFile))
```

Comment:
Why do you save as a file instead of passing the photo directly ?

Reply:
Yes, it would be better to only save the photo (like WhatsApp, Telegram, ... do) once it has been confirmed that the photo will be posted. Or do you think we should not save it at all?

Reply:
Not sure, we can leave it as it is for now and see with Aman what the best way is for posting photos

Reply:
Yes, I can take care of integrating this with the upload in my next sprint.